charts/minecraft-bedrock/README.md

```diff
@@ -0,0 +1,78 @@
+# Minecraft
+
+[Minecraft](https://minecraft.net/en/) is a game about placing blocks and going on adventures.
+
+## Introduction
+
+This chart creates a single Minecraft Pod, plus Services for the Minecraft server and RCON.
+
+## Prerequisites
+
+- 512 MB of RAM
+- Kubernetes 1.4+ with Beta APIs enabled
+- PV provisioner support in the underlying infrastructure
+
+## Installing the Chart
+
+To install the chart with the release name `minecraft`, read the [Minecraft EULA](https://account.mojang.com/documents/minecraft_eula) run:
+
+```shell
+helm install minecraft-bedrock \
+  --set minecraftServer.eula=true itzg/minecraft
+```
+
+This command deploys a Minecraft dedicated server with sensible defaults.
+
+> **Tip**: List all releases using `helm list`
+
+## Uninstalling the Chart
+
+To uninstall/delete the `minecraft-bedrock` deployment:
+
+```shell
+helm delete minecraft-bedrock
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+Refer to [values.yaml](values.yaml) for the full run-down on defaults. These are a mixture of Kubernetes and Minecraft-related directives that map to environment variables in the [itzg/minecraft-bedrock-server](https://hub.docker.com/r/itzg/minecraft-bedrock-server/) Docker image.
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```shell
+helm install --name minecraft-bedrock \
+  --set minecraftServer.eula=true,minecraftServer.Difficulty=hard \
+  itzg/minecraft
+```
+
+Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example,
+
+```shell
+helm install --name minecraft -f values.yaml itzg/minecraft
+```
+
+> **Tip**: You can use the default [values.yaml](values.yaml)
+
+## Persistence
+
+The [itzg/minecraft-bedrock-server](https://hub.docker.com/r/itzg/minecraft-bedrock-server/) image stores the saved games and mods under /data.
+
+By default a PersistentVolumeClaim is created and mounted for saves but not mods. In order to disable this functionality
+you can change the values.yaml to disable persistence under the sub-sections under `persistence`.
+
+> *"An emptyDir volume is first created when a Pod is assigned to a Node, and exists as long as that Pod is running on that node. When a Pod is removed from a node for any reason, the data in the emptyDir is deleted forever."*
+
+## Backups
+
+You can backup the state of your minecraft server to your local machine via the `kubectl cp` command.  
+
+```shell
+NAMESPACE=default
+POD_ID=lionhope-387ff8d-sdis9
+kubectl exec --namespace ${NAMESPACE} ${POD_ID} rcon-cli save-off
+kubectl exec --namespace ${NAMESPACE} ${POD_ID} rcon-cli save-all
+kubectl cp ${NAMESPACE}/${POD_ID}:/data .
+kubectl exec --namespace ${NAMESPACE} ${POD_ID} rcon-cli save-on
```

Comment:
```shell
NAMESPACE=default
POD_ID=lionhope-387ff8d-sdis9
kubectl attach --namespace ${NAMESPACE} ${POD_ID} 
save hold
save query
kubectl cp ${NAMESPACE}/${POD_ID}:/data .
kubectl attach --namespace ${NAMESPACE} ${POD_ID} save resume
```

This is how I would image it would go, but it seems I don't get a console prompt to even be able to run a command. Even running this just stand alone on docker and following the steps you mentioned [here](https://github.com/itzg/docker-minecraft-bedrock-server#executing-server-commands) (I am not an expert at minecraft servers so I might just be missing something) I am unable to actually see any changes on the server or get any response.

Again, I am not an expert, but could we take advantage of pvc snapshotting or something similar to back up the world or does minecraft require these steps to properly back up the world?

Reply:
With a majority of stateful workloads, especially databases, there is a slight risk of snapshotting inconsistent file state. As such, there is usually a method to flush-and-pause file operations or export backups in a consistent manner. The bedrock server appears to do a version of the latter for backups to ensure that the world data files are consistent during a backup/snapshot.

The comment I added about `stdin` and `tty` will resolve the issue you saw with attaching.